pgjdbc/src/test/java/org/postgresql/test/jdbc42/SetObject310Test.java

```diff
@@ -160,7 +161,7 @@ private void insertWithoutType(Object data, String columnName) throws SQLExcepti
         assertNotNull(rs);
 
         assertTrue(rs.next());
-        return expectedType.cast(rs.getObject(1));
+        return type.cast(rs.getObject(1, type));
```

Comment:
Why change this?
I think this was deliberately added to test `getObject(int)` behavior, and adding another argument changes the code path.

It is important that the driver does not accidentally change `getObject(int)` result even though there's no exact specification on what `getObject` should return.

Reply:
This change actuall verifies that setObject/getObject is symmetric. I know that generic getObject() is not needed.

I will change this to actually do two tests to check both variants.
- The return as done here should be reverted, YES!
- Before retuning the generic Object value, the test should assert that "Object data" and the value returned by `rs.getObject(1, data.getClass()` are equal.

When doing this we check both (symmetry - which is way more important) and that the untyped `getObject()` returns the input unmodified.

Reply:
I fixed this, see my last commit. The latest change brings both!